### ISSUE TITLE
Add option to set quality of screenshot

### ISSUE THREAD

**Issue by Jacalz** [lupoDharkael/flameshot#506]
Right now we get what looks like quite hard compressed images which du get small in size but they don’t look all that good. It would be great to have a slider where  you can set the compression strength and possible choose to don’t compress the images.

**Comment by Mershl** [comment]
Flameshot is using the screenshot interface of the respective WindowManager you're using (see src/utils/screengrabber.cpp).

Based on the screenshot interface of Gnome I don't see an option to improve the quality ([org.gnome.Shell.Screenshot](https://github.com/vinzenz/gnome-shell/blob/master/data/org.gnome.Shell.Screenshot.xml)).

Gnome's onboard screenshot tool is suffering of the same compression artifacts.

**Comment by Jacalz** [comment]
Weird. I thought Gnome Screenshots gave higher quality screenshots, but it cool to get more understanding of how the software works. Thank you :+1:

**Comment by ganomi46** [comment]
I have got also picture quality issue in KDE Debian. It would be great if we can set picture quality.

**Comment by crimsonskyrem** [comment]
I got this issue, too
my screenshot image quality was very low that make it too hard to see
then I found that's because I deleted the default gnome-screenshot
after I reinstalled the gnome-screenshot, the image quality was good to see

**Comment by thecodemonk** [comment]
I don't know as if this is correct.. I'm on Pop OS and my screenshots with the built in gnome screenshot are WAY better quality than Flameshot. I found this issue after searching on how to fix it. The screenshots with Flameshot are compressed way more than the built in gnome-screenshot.

**Comment by pymen** [comment]
still actul!

**Comment by Martin-Eckleben** [comment]
Recently someone mentioned KDE < 5.19  + Wayland will compress screenshots as well.
Ill look into this but it will probably take some time because this repo has a lot to catch up to.

**Comment by pymen** [comment]
Looks like quality can be set via https://doc.qt.io/qt-5/qpixmap.html  (i am not a c dev)

**Comment by borgmanJeremy** [comment]
@pymen What screen resolution do you have? I am looking to reproduce the issue.

**Comment by pymen** [comment]
> @pymen What screen resolution do you have? I am looking to reproduce the issue.

I do not have an issue, I just wanted to say that it will be nice to have some configuration to set the quality of the capture image, for example, if I need to post IMG to slack I do not need quality=10, 3 will be enough

**Comment by borgmanJeremy** [comment]
Ohh ok, I was looking into an issue where people are saying Flameshot is compressing the image when they do not want it to.

**Comment by funnisquares** [comment]
This option would be really good. I'm using AwesomeWm with Plasma and the compression is really apparent.

**Comment by mmahmoudian** [comment]
@RJN-bot what version if Flameshot are you using? In general PNG is a lossless compressed image format and you should not see any quality issues with PNG. So make sure you are saving as PNG and not JPEG.

**Comment by YSCohen** [comment]
I also have this problem. I'm using scoop-installed Flameshot on Windows 10. As you can see, the below image is very blurry. It has the same resolution as a similar shot taken with win+shift+s
![image](https://user-images.githubusercontent.com/78832113/158092772-669f9527-f610-4b5d-9a4a-2259c4afb8d8.png)

**Comment by hubyhuby** [comment]
Please include WEBP format . It is a very good open image format for the web.

**Comment by mauromotion** [comment]
How is this closed? 

I'm using Flameshot on Arch with i3. I want to use the JPG compression but I need to be able to set the compression level like in 99.999% of other screenshot software out there. If different file formats exist there are several reasons, so don't just tell people "use PNG". If someone needs to use JPG they also usually need to be able to set its compression (the main feature of the JPG is indeed image compression while PNG's isn't). Please don't let me get back to scrot...at least set the default JPG quality to a higher level, as it's not very usable right now (try to get a screenshot of some text and you'll see). Thanks.

**Comment by mmahmoudian** [comment]
@mauromotion you wrote two points and here is the answer:

1. About this issue being closed: please pay attention before pointing fingers

![aa](https://github.com/flameshot-org/flameshot/assets/390889/0e74cfe1-c02e-40eb-b390-402a9d5a5ade)

2. If you want lossless, don't use a lossy format. That simple. You are free to choose whatever software you want, but the reality is that JPEG is a lossy method by design. It meant to be lossy.

**Comment by mauromotion** [comment]
Yeah my fault for not noticing it was still open. I'm still new to GitHub.

But still, it's very limiting not being able to select the level of compression of a lossy format. Who's deciding for me? And on what basis?

All I can see is that whoever decided took a wrong choice because if  I take a screenshot with Flameshot as a JPG the resulting image is unusable, forcing me to choose another file format. Why do I have to use a much larger PNG file when I know for a fact that if I could set the compression level of the JPG it would be just fine? 
Not being able to set the compression level of a JPG file completely hinders the usage of a JPG file format, IMO.

**Comment by PoneyClairDeLune** [comment]
Would be delighted to see Flameshot offering WebP (`q` & `psnr`) and JPEG XL (`d`) support with quality controls.

**Comment by krlabs** [comment]
Webp is what you need to be happy. We often take screenshots for publications on the site. And then we convert them to WebP separately, because this is the most optimised format for images on the web. If Flameshot added this feature, it would be priceless.

**Comment by PoneyClairDeLune** [comment]
@krlabs #3672

**Comment by Wylbur** [comment]
> WebP is what you need to be happy. We often take screenshots for publications on the site. And then we convert them to WebP separately, because this is the most optimised format for images on the web. If Flameshot added this feature, it would be priceless.

For anyone landing here that is looking for file format support on Linux, please note. I installed the latest Linux download https://github.com/flameshot-org/flameshot/releases/download/v13.3.0/flameshot-13.3.0-1.ubuntu-24.04.amd64.deb. This version does NOT support many file formats that are available with the same version of the flatpak version! There are 26 formats available via the flatpak version and less than a dozen with the .deb version.

For me it was specifically missing the .webp extension.

Not sure why the versions are so far apart! Hope this helps someone else looking for options!
